src/parser/util.js

```diff
@@ -79,6 +79,6 @@ pp.expect = function (type) {
 
 // Raise an unexpected token error.
 
-pp.unexpected = function (pos) {
-  this.raise(pos != null ? pos : this.state.start, "Unexpected token");
+pp.unexpected = function (pos, message = "Unexpected token") {
+  this.raise(pos != null ? pos : this.state.start, message);
```

Comment:
I like being able to change the error message. Unexpected token is not fun to see (my example is `function a() { await b; }`) without async in front of function